app/controllers/api/v2/os_default_templates_controller.rb

```diff
@@ -63,12 +65,19 @@ def destroy
       private
 
       def rename_config_template
-        if !params[:config_template_id].nil?
+        if params[:config_template_id].present?
           params[:provisioning_template_id] = params.delete(:config_template_id)
           Foreman::Deprecation.api_deprecation_warning("Config templates were renamed to provisioning templates")
         end
       end
 
+      def rename_nested_config_template
+        if params[:os_default_template] && params[:os_default_template][:config_template_id].present?
```

Comment:
I think copying it is sufficient though, because we only ever use the nested version in create/update:

    @os_default_template = nested_obj.os_default_templates.new(params[:os_default_template])

I get this now at the top of rename_config_template which looks correct:

<pre>
[1] pry(#&lt;Api::V2::OsDefaultTemplatesController>)> params
=> {"template_kind_id"=>1,
 "template_kind_name"=>"PXELinux",
 "config_template_id"=>1,
 "config_template_name"=>"Kickstart default PXELinux",
 "format"=>"json",
 "apiv"=>"v2",
 "action"=>"create",
 "controller"=>"api/v2/os_default_templates",
 "operatingsystem_id"=>"6",
 "os_default_template"=>{"template_kind_id"=>1, "config_template_id"=>1}}
</pre>

So I don't think the unnested copy is required.